src/main/java/org/isf/medicalstock/rest/StockMovementController.java

```diff
@@ -0,0 +1,247 @@
+package org.isf.medicalstock.rest;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+import org.isf.medicals.manager.MedicalBrowsingManager;
+import org.isf.medicals.model.Medical;
+import org.isf.medicalstock.dto.LotDTO;
+import org.isf.medicalstock.dto.MovementDTO;
+import org.isf.medicalstock.manager.MovBrowserManager;
+import org.isf.medicalstock.manager.MovStockInsertingManager;
+import org.isf.medicalstock.mapper.LotMapper;
+import org.isf.medicalstock.mapper.MovementMapper;
+import org.isf.medicalstock.model.Lot;
+import org.isf.medicalstock.model.Movement;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.isf.ward.model.Ward;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/stockmovements", produces = MediaType.APPLICATION_JSON_VALUE)
+public class StockMovementController {
+	@Autowired
+	private MovementMapper movMapper;
+	
+	@Autowired
+	private LotMapper lotMapper;
+	
+	@Autowired
+	private MovBrowserManager movManager;
+	
+	@Autowired
+	private MovStockInsertingManager movInsertingManager;
+	
+	@Autowired
+	private MedicalBrowsingManager medicalManager;
+	
+	/**
+	 * Insert a list of charging {@link Movement}s and related {@link Lot}s
+	 * 
+	 * @param movementDTOs - the list of {@link Movement}s
+	 * @param referenceNumber - the reference number to be set for all movements
+	 * 		   if {@link null}, each movements must have a different referenceNumber 
+	 * @return 
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/stockmovements/charge", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> newMultipleChargingMovements(@RequestBody List<MovementDTO> movementDTOs, 
+			@RequestParam(name="ref", required=true) String referenceNumber) throws OHServiceException {
+		ArrayList<Movement> movements = new ArrayList<Movement>();
+		movMapper.map2ModelList(movementDTOs).forEach(mov -> {
+			movements.add(mov);
```

Comment:
We probably avoid foreach iteration assigning movMapper.map2ModelList(movementDTOs) to movements?

Reply:
The method of the manager is waiting for an ArrayList of Movement. I have prefered to manually construct the ArrayList to pass to the manager's method instead of casting the List returned by the map2ModelList method of the mapper to an ArrayList.